### PR TITLE
[WIP] ENH: Add --func-only option to skip anatomical processing

### PR DESCRIPTION
**Pull request by madeleinesnyder**
- [x] Connect [this workflow](https://github.com/oesteban/fmriprep/blob/b78a50dfb08483aba99d90292e1b8367d8440222/fmriprep/workflows/base.py#L459) or, probably, return it instead of `wf`.
  - [x] Fill up the structure of the new workflow [`init_onlyfunc_preproc_wf`](https://github.com/oesteban/fmriprep/blob/b78a50dfb08483aba99d90292e1b8367d8440222/fmriprep/workflows/bold/base.py#L571)
  - [ ] Write func-only alternatives to the Summary node and the Reports node, and connect them.
  - [x] Verify your name, affiliation and insert your ORCID (if available) in the [credits file](https://github.com/oesteban/fmriprep/blob/16c367a17d5c14030a443c780443242a315068c8/.zenodo.json#L17-L20).

### CODE REVIEW

**Comment by madeleinesnyder** [comment]
Ah yes, got it. I see now there is a whole new workflow to take care of the func-only case instead of editing the original func workflow.

Also, I was thinking that it might be helpful for me to contribute a BIDSgenerator file that would be easily configured by people who want to use the pipeline but don't have the BIDS data format. I know our lab was hesitant to use the pipeline because of the overhead of renameing and restructuring data, so it might increase use if there is an add-on file that takes care of this. I could put in the places in run.py and other scripts where BIDSgenerator.py is called

**Comment by oesteban** [comment]
Sure, but it is a lot easier to keep things focused :). Feel free to open as many PRs as features you'd like to contribute. It's much more manageable when contributions are handled separatelly

**Comment by oesteban** [comment]
As you can see above, I just pushed one commit to your branch. You can see it where it says "convert indentation to spaces" above.

To get those changes, make sure you `git pull` on your repo after you make sure you are in the correct branch with `git checkout func-only`.

**Comment by madeleinesnyder** [comment]
Ok, so I don't mess this up:

To open a PR to push the BIDSgenerator.py file and other files updated with (commented out) lines to handle this script, I would do a new PR from the https://github.com/poldracklab/fmriprep page on the master branch?

**Comment by oesteban** [comment]
Yes - to open a new PR:

1. Make sure you are in sync with the upstream repo:

```
git fetch upstream
git checkout master
git merge upstream/master
```

2. Open up a new branch

```
git checkout -b new-feature-branch
```

3. Edit the code (add changes now), commit changes and push:

```
<edit files>
git add <new files that were not in the repo already>
git commit -a
git push --set-upstream origin new-feature-branch
```
4. Go to GitHub.com and request a new PR

5. Repeat 3 until the code is acceptable to be merged in.

**Comment by madeleinesnyder** [comment]
Alrighty I just requested a new one-- it should be just two files, one added (BIDSgenerator.py) and one modified (run.py)

**Comment by oesteban** [comment]
Cool!

Let's focus on getting this one merged first :+1: 

I have updated your PR comment up above with the tasks that need be done before merging.

You will also see below my comment a message saying "All checks have failed". This means that there are problems with the code and it didn't pass tests. Let's not worry about this now, but tests will need to get green before we can merge. We'll address this in a future.

Please let me know what is stopping you at this point (if any).

**Comment by madeleinesnyder** [comment]
Ok sounds good. I just connected up the onlyfunc workflow in base.py and I
need to know what to merge:
I did
git add --all
git commit -m "[my message]"
now what? (git push --set-upstream origin master?)

**Comment by oesteban** [comment]
```
git push
```

The `--set-upstream origin` is only for the first time you push. After that origin is always set when you push.

**Comment by madeleinesnyder** [comment]
ok done

**Comment by oesteban** [comment]
Indeed!

I can see your changes here: https://github.com/poldracklab/fmriprep/pull/808/commits/940340b6e630eb27b0c0673b54653531d3309404

BTW, with the `git add --all` you also included a file called `fmriprep/workflows/.base.py.swo` (vim's temporary files).

Please remove it:
```
git rm fmriprep/workflows/.base.py.swo
git commit -am 'remove fmriprep/workflows/.base.py.swo'
git push
```

File: .zenodo.json [hunk 1]
```diff
@@ -15,6 +15,10 @@
       "orcid": "0000-0002-6533-164X"
     },
     {
+      "name": "Madeleine Snyder",
+      "affiliation": "Harvard Medical School, Harvard University, Cambridge, Massachusetts"
+    },
+    {
```
Comment (effigies) [inline .zenodo.json]: Could you move this to just above Russ Poldrack? Outside contributors are listed in order of contribution, starting at Feilong Ma.

Also, we haven't generally included city and state in the affiliation. For consistency, I'd suggest dropping it. And do you have an [ORCID](https://orcid.org/)? It doesn't take much time to get one, and they help disambiguate researchers with similar names, as well as persist across changes in institution/name.

File: fmriprep/workflows/base.py [hunk 2]
```diff
@@ -374,77 +382,113 @@ def init_single_subject_wf(subject_id, task_id, name,
...
+                                        ('t2w', 'inputnode.t2w')]),
+            (summary, anat_preproc_wf, [('subject_id', 'inputnode.subject_id')]),
+            (bidssrc, ds_summary_report, [(('t1w', fix_multi_T1w_source_name), 'source_file')]),
+            (summary, ds_summary_report, [('out_report', 'in_file')]),
+            (bidssrc, ds_about_report, [(('t1w', fix_multi_T1w_source_name), 'source_file')]),
+            (about, ds_about_report, [('out_report', 'in_file')]),
+        ])
+
+        if anat_only:
+            return workflow
```
Comment (effigies) [inline fmriprep/workflows/base.py]: If I specify `--anat-only` and `--func-only`, this won't be hit, so the functional workflows will still be generated. There should probably at some point (possibly in `run.py`) be a check to raise an exception if both flags are passed to fmriprep.

Comment (madeleinesnyder) [reply fmriprep/workflows/base.py]: Ok, I would just put in a simple if anat_only and func_only raise exception before any of the workflows are put together. Any reasons to put it elsewhere?

Comment (oesteban) [reply fmriprep/workflows/base.py]: The "proper" way to do this is using [mutually exclusive options](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group)

```Python
    # Create a mutually exclusive group for --anat-only and --func-only
    g_blocks = parser.add_mutually_exclusive_group()
    g_blocks.add_argument('--anat-only', action='store_true',
                          help='run anatomical workflows only')
    g_blocks.add_argument('--func-only', action='store_true',
                          help='run functional workflows only '
                               '(dismiss the anatomical images if present).')
```

File: .zenodo.json (hunk 1 above)
Comment (oesteban) [reply .zenodo.json]: Sorry for the ordering issue, it's my bad. @madeleinesnyder please fix the ordering and add your ORCID if available/possible.

File: fmriprep/workflows/base.py [hunk 3]
```diff
@@ -373,78 +381,116 @@ def init_single_subject_wf(subject_id, task_id, name,
...
+                fmap_demean=fmap_demean,
+                debug=debug,
+                use_aroma=use_aroma,
+                ignore_aroma_err=ignore_aroma_err,
+            )
+
+            workflow.connect([
+                (inputnode, func_preproc_wf, [('subjects_dir', 'inputnode.subjects_dir')]),
+                (bidssrc, bids_info, [
+                    (('bold', fix_multi_bold_source_name), 'in_file')]),
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: `fix_multi_bold_source_name` is undefined

File: fmriprep/workflows/base.py [hunk 4]
```diff
@@ -373,78 +381,116 @@ def init_single_subject_wf(subject_id, task_id, name,
...
+                (inputnode, summary, [('subjects_dir', 'subjects_dir')]),
+                (bidssrc, summary, [('bold', 'bold')]),
+                (bids_info, summary, [('subject_id', 'subject_id')]),
+                (bidssrc, func_preproc_wf, [('bold', 'inputnode.bold')]),
+                (summary, func_preproc_wf, [('subject_id', 'inputnode.subject_id')]),
+                (bidssrc, ds_summary_report, [
+                    (('bold', fix_multi_bold_source_name), 'source_file')]),
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: `fix_multi_bold_source_name` is undefined

File: fmriprep/workflows/base.py [hunk 5]
```diff
@@ -373,78 +381,116 @@ def init_single_subject_wf(subject_id, task_id, name,
...
+                (summary, ds_summary_report, [('out_report', 'in_file')]),
+                (bidssrc, ds_about_report, [
+                    (('bold', fix_multi_bold_source_name), 'source_file')]),
```
Comment (oesteban) [inline fmriprep/workflows/base.py]: `fix_multi_bold_source_name` is undefined

**Comment by oesteban** [comment]
I can put some hours on pushing this PR forward this week, WDYT @madeleinesnyder ?

There are some outstanding errors, if you look at [Travis tests](https://travis-ci.org/poldracklab/fmriprep/builds/301573935?utm_source=github_status&utm_medium=notification):

```
$ flake8 fmriprep wrapper
fmriprep/workflows/base.py:482:31: F821 undefined name 'fix_multi_bold_source_name'
fmriprep/workflows/base.py:489:31: F821 undefined name 'fix_multi_bold_source_name'
fmriprep/workflows/base.py:492:31: F821 undefined name 'fix_multi_bold_source_name'
fmriprep/workflows/bold/base.py:751:9: F841 local variable 'bold_pe' is assigned to but never used
fmriprep/workflows/bold/base.py:772:5: E303 too many blank lines (2)
fmriprep/workflows/bold/base.py:779:9: F841 local variable 'bold_stc_wf' is assigned to but never used
fmriprep/workflows/bold/base.py:788:9: F821 undefined name 'summary'
fmriprep/workflows/bold/base.py:825:29: F821 undefined name 'func_reports_wf'
fmriprep/workflows/bold/base.py:843:9: E303 too many blank lines (2)
fmriprep/workflows/bold/base.py:845:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:846:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:848:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:848:23: F821 undefined name 'bold_confounds_wf'
fmriprep/workflows/bold/base.py:848:97: W291 trailing whitespace
fmriprep/workflows/bold/base.py:849:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:849:29: F821 undefined name 'func_reports_wf'
fmriprep/workflows/bold/base.py:851:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:851:10: F821 undefined name 'bold_confounds_wf'
fmriprep/workflows/bold/base.py:856:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:856:10: F821 undefined name 'bold_confounds_wf'
fmriprep/workflows/bold/base.py:856:29: F821 undefined name 'func_reports_wf'
fmriprep/workflows/bold/base.py:860:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:860:10: F821 undefined name 'bold_confounds_wf'
fmriprep/workflows/bold/base.py:860:29: F821 undefined name 'summary'
fmriprep/workflows/bold/base.py:861:9: E122 continuation line missing indentation or outdented
fmriprep/workflows/bold/base.py:861:10: F821 undefined name 'summary'
fmriprep/workflows/bold/base.py:861:19: F821 undefined name 'func_reports_wf'
```

File: fmriprep/workflows/base.py (hunk 3 above)
Comment (madeleinesnyder) [reply fmriprep/workflows/base.py]: I assumed there was an analog to fic_multi_t1w_source_name, which pops up in the anat_only workflow.

**Comment by madeleinesnyder** [comment]
Yes, I can come work on main campus if that would help. Or work remotely from my lab. Either works. Let me know.

**Comment by oesteban** [comment]
Hey @madeleinesnyder, we released [fmriprep-1.0.0](https://github.com/poldracklab/fmriprep/releases/tag/1.0.0) yesterday. I am hoping that after next week I'll have more time and can help you finish this up. Let me know when you can touch base.

**Comment by madeleinesnyder** [comment]
Congrats! My schedule is pretty flexible all next week, so let me know what
time works best for you.

**Pull request closed without merging**